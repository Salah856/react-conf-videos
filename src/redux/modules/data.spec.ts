import 'rxjs'
import dataReducer, {
  transformDataFromJson,
  initialState,
  cleanString,
  COPY_DATA
} from './data'

const inputJSON = require('../../../scripts/testInput.json')

describe('data', () => {

  describe('reducer', () => {
    it('should return the intial state', () => {
      // arrange
      const action = { type: 'EEE' }

      // act
      const result = dataReducer(undefined, action)

      // expect
      expect(result).toBe(initialState)
    })

    it('should copy data to the slice', () => {
      // arrange
      const payload = initialState.merge({ presenters: { idx: { name: 'lalala' } } })
      const action = { type: COPY_DATA, payload }

      // act
      const result = dataReducer(undefined, action)

      // expect
      expect(result).toEqual(payload)
    })
  })

  describe('cleanString', () => {
    it('should lowercase all chars', () => {
      // arrange
      const input = 'AAAAAAAA'
      const expectedResult = 'aaaaaaaa';

      // act
      const result = cleanString(input)

      // assert
      expect(result).toEqual(expectedResult)
    })

    it('should lowercase & normalize all diacritical chars', () => {
      // arrange
      const input = 'ÄÖÜÑ'
      const expectedResult = 'aoun';

      // act
      const result = cleanString(input)

      // assert
      expect(result).toEqual(expectedResult)
    })
  })

  describe('transformDataFromJson', () => {
    it('should match the snapshot', () => {
      // arrange
      // act
      const result = transformDataFromJson(inputJSON)

      // assert
      expect(result).toMatchSnapshot()
    })
  })

})
