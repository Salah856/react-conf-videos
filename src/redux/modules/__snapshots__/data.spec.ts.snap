// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data transformDataFromJson should match the snapshot 1`] = `
Object {
  "conferences": Object {
    "JTIyUmVhY3QlMjBDb25mJTIwMjAxNyUyMg==": Object {
      "date": "13-03-2017",
      "playlist": "https://www.youtube.com/playlist?list=PLb0IAmt7-GS3fZ46IGFirdqKTIxlws7e0",
      "title": "React Conf 2017",
      "videos": Array [
        "JTIyS2V5bm90ZSUyMFBhcnQlMjAxJTIy",
        "JTIyS2V5bm90ZSUyMFBhcnQlMjAyJTIwLSUyMEluY3JlbWVudGFsbHklMjBBZG9wdGluZyUyMFJlYWN0JTIwTmF0aXZlJTIwYXQlMjBGYWNlYm9vayUyMg==",
      ],
      "website": "http://conf.reactjs.org/",
    },
    "JTIyUmVhY3QlMjBSYWxseSUyMDIwMTclMjI=": Object {
      "date": "24-08-2017",
      "playlist": "https://www.youtube.com/watch?v=2rhkgB8Cohc&list=PLUqJuYVCz-7eHxPEPz8hJCz2pW6N0H531",
      "title": "React Rally 2017",
      "videos": Array [
        "JTIyVU5QS0clM0ElMjBUaGUlMjBDRE4lMjBmb3IlMjBldmVyeXRoaW5nJTIwb24lMjBucG0lMjI=",
        "JTIyRDMlMjAlMjYlMjBSZWFjdCUyMg==",
      ],
      "website": "http://www.reactrally.com/",
    },
    "JTIyUmVhY3RpdmVDb25mJTIwMjAxNiUyMg==": Object {
      "date": "26-10-2016",
      "playlist": "https://www.youtube.com/watch?v=gbATtWPUfN0&list=PLa2ZZ09WYepMCRRGCRPhTYuTCat4TiDlX",
      "title": "ReactiveConf 2016",
      "videos": Array [
        "JTIyV3JpdGluZyUyMFNvdW5kJTIwQXN5bmNocm9ub3VzJTIwQ29kZSUyMHdpdGglMjBDb3JvdXRpbmVzJTIy",
      ],
      "website": "https://reactiveconf.com/2016/",
    },
  },
  "presenters": Object {
    "JTIySmluZyUyMENoZW4lMjI=": Object {
      "name": "Jing Chen",
    },
    "JTIyTWljaGFlbCUyMEphY2tzb24lMjI=": Object {
      "name": "Michael Jackson",
    },
    "JTIyU2hpcmxleSUyMFd1JTIy": Object {
      "name": "Shirley Wu",
    },
    "JTIyVG9tJTIwT2NjaGlubyUyMg==": Object {
      "name": "Tom Occhino",
    },
    "JTIyVG9tYSVDNSVBMSUyMEt1bGljaCUyMg==": Object {
      "name": "Toma≈° Kulich",
    },
  },
  "presentersSearchable": Object {
    "JTIySmluZyUyMENoZW4lMjI=": Object {
      "name": "jing chen",
    },
    "JTIyTWljaGFlbCUyMEphY2tzb24lMjI=": Object {
      "name": "michael jackson",
    },
    "JTIyU2hpcmxleSUyMFd1JTIy": Object {
      "name": "shirley wu",
    },
    "JTIyVG9tJTIwT2NjaGlubyUyMg==": Object {
      "name": "tom occhino",
    },
    "JTIyVG9tYSVDNSVBMSUyMEt1bGljaCUyMg==": Object {
      "name": "tomas kulich",
    },
  },
  "videos": Object {
    "JTIyRDMlMjAlMjYlMjBSZWFjdCUyMg==": Object {
      "embeddableLink": "https://www.youtube.com/embed/ladXdJ3KKd4",
      "length": "26:57",
      "link": "https://www.youtube.com/watch?v=ladXdJ3KKd4&index=2&list=PLUqJuYVCz-7eHxPEPz8hJCz2pW6N0H531",
      "presenter": "JTIyU2hpcmxleSUyMFd1JTIy",
      "split": "Day 1",
      "title": "D3 & React",
    },
    "JTIyS2V5bm90ZSUyMFBhcnQlMjAxJTIy": Object {
      "embeddableLink": "https://www.youtube.com/embed/7HSd1sk07uU",
      "length": "6:36",
      "link": "https://www.youtube.com/watch?v=7HSd1sk07uU&index=1&list=PLb0IAmt7-GS3fZ46IGFirdqKTIxlws7e0",
      "presenter": "JTIyVG9tJTIwT2NjaGlubyUyMg==",
      "split": "Day 1",
      "title": "Keynote Part 1",
    },
    "JTIyS2V5bm90ZSUyMFBhcnQlMjAyJTIwLSUyMEluY3JlbWVudGFsbHklMjBBZG9wdGluZyUyMFJlYWN0JTIwTmF0aXZlJTIwYXQlMjBGYWNlYm9vayUyMg==": Object {
      "embeddableLink": "https://www.youtube.com/embed/cSUxHv-kH7w",
      "length": "14:13",
      "link": "https://www.youtube.com/watch?v=cSUxHv-kH7w&list=PLb0IAmt7-GS3fZ46IGFirdqKTIxlws7e0&index=2",
      "presenter": "JTIySmluZyUyMENoZW4lMjI=",
      "split": "Day 1",
      "title": "Keynote Part 2 - Incrementally Adopting React Native at Facebook",
    },
    "JTIyV3JpdGluZyUyMFNvdW5kJTIwQXN5bmNocm9ub3VzJTIwQ29kZSUyMHdpdGglMjBDb3JvdXRpbmVzJTIy": Object {
      "embeddableLink": "https://www.youtube.com/embed/gZz-1OaMpqg",
      "length": "35:43",
      "link": "https://www.youtube.com/watch?v=gZz-1OaMpqg&index=29&list=PLa2ZZ09WYepMCRRGCRPhTYuTCat4TiDlX",
      "presenter": "JTIyVG9tYSVDNSVBMSUyMEt1bGljaCUyMg==",
      "split": "Day 2",
      "title": "Writing Sound Asynchronous Code with Coroutines",
    },
    "JTIyVU5QS0clM0ElMjBUaGUlMjBDRE4lMjBmb3IlMjBldmVyeXRoaW5nJTIwb24lMjBucG0lMjI=": Object {
      "embeddableLink": "https://www.youtube.com/embed/2rhkgB8Cohc",
      "length": "31:46",
      "link": "https://www.youtube.com/watch?v=2rhkgB8Cohc&list=PLUqJuYVCz-7eHxPEPz8hJCz2pW6N0H531",
      "presenter": "JTIyTWljaGFlbCUyMEphY2tzb24lMjI=",
      "split": "Day 1",
      "title": "UNPKG: The CDN for everything on npm",
    },
  },
  "videosSearchable": Object {
    "JTIyRDMlMjAlMjYlMjBSZWFjdCUyMg==": Object {
      "embeddableLink": "https://www.youtube.com/embed/ladXdJ3KKd4",
      "length": "26:57",
      "link": "https://www.youtube.com/watch?v=ladXdJ3KKd4&index=2&list=PLUqJuYVCz-7eHxPEPz8hJCz2pW6N0H531",
      "presenter": "JTIyU2hpcmxleSUyMFd1JTIy",
      "split": "day 1",
      "title": "d3 & react",
    },
    "JTIyS2V5bm90ZSUyMFBhcnQlMjAxJTIy": Object {
      "embeddableLink": "https://www.youtube.com/embed/7HSd1sk07uU",
      "length": "6:36",
      "link": "https://www.youtube.com/watch?v=7HSd1sk07uU&index=1&list=PLb0IAmt7-GS3fZ46IGFirdqKTIxlws7e0",
      "presenter": "JTIyVG9tJTIwT2NjaGlubyUyMg==",
      "split": "day 1",
      "title": "keynote part 1",
    },
    "JTIyS2V5bm90ZSUyMFBhcnQlMjAyJTIwLSUyMEluY3JlbWVudGFsbHklMjBBZG9wdGluZyUyMFJlYWN0JTIwTmF0aXZlJTIwYXQlMjBGYWNlYm9vayUyMg==": Object {
      "embeddableLink": "https://www.youtube.com/embed/cSUxHv-kH7w",
      "length": "14:13",
      "link": "https://www.youtube.com/watch?v=cSUxHv-kH7w&list=PLb0IAmt7-GS3fZ46IGFirdqKTIxlws7e0&index=2",
      "presenter": "JTIySmluZyUyMENoZW4lMjI=",
      "split": "day 1",
      "title": "keynote part 2 - incrementally adopting react native at facebook",
    },
    "JTIyV3JpdGluZyUyMFNvdW5kJTIwQXN5bmNocm9ub3VzJTIwQ29kZSUyMHdpdGglMjBDb3JvdXRpbmVzJTIy": Object {
      "embeddableLink": "https://www.youtube.com/embed/gZz-1OaMpqg",
      "length": "35:43",
      "link": "https://www.youtube.com/watch?v=gZz-1OaMpqg&index=29&list=PLa2ZZ09WYepMCRRGCRPhTYuTCat4TiDlX",
      "presenter": "JTIyVG9tYSVDNSVBMSUyMEt1bGljaCUyMg==",
      "split": "day 2",
      "title": "writing sound asynchronous code with coroutines",
    },
    "JTIyVU5QS0clM0ElMjBUaGUlMjBDRE4lMjBmb3IlMjBldmVyeXRoaW5nJTIwb24lMjBucG0lMjI=": Object {
      "embeddableLink": "https://www.youtube.com/embed/2rhkgB8Cohc",
      "length": "31:46",
      "link": "https://www.youtube.com/watch?v=2rhkgB8Cohc&list=PLUqJuYVCz-7eHxPEPz8hJCz2pW6N0H531",
      "presenter": "JTIyTWljaGFlbCUyMEphY2tzb24lMjI=",
      "split": "day 1",
      "title": "unpkg: the cdn for everything on npm",
    },
  },
}
`;
